File: app/src/main/java/co/edu/unipiloto/myapplication/adapters/DashboardPagerAdapter.java
package co.edu.unipiloto.myapplication.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import co.edu.unipiloto.myapplication.fragment.AssignedRequestsFragment;
import co.edu.unipiloto.myapplication.ui.*; // Asegúrate de que este import abarque BranchPendingFragment

public class DashboardPagerAdapter extends FragmentStateAdapter {

    private static final int NUM_TABS = 2;

    public DashboardPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                // 🏆 CORRECCIÓN: Usar la clase que ya existe (BranchPendingFragment)
                // Carga la lista de solicitudes sin asignar
                return new BranchPendingFragment();
            case 1:
                // Carga la lista de solicitudes ya asignadas
                // Asumimos que esta clase sí existe o la crearemos después.
                return new AssignedRequestsFragment();
            default:
                return new BranchPendingFragment();
        }
    }

    @Override
    public int getItemCount() {
        return NUM_TABS;
    }
}

File: app/src/main/java/co/edu/unipiloto/myapplication/adapters/DashboardPagerAdapter.kt
package co.edu.unipiloto.myapplication.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import co.edu.unipiloto.myapplication.fragment.PendingRequestsFragment
import co.edu.unipiloto.myapplication.fragment.AssignedRequestsFragment

/**
 * 📊 Adaptador para ViewPager2 que gestiona las pestañas del Panel de Control del Gerente/Funcionario.
 * Carga un Fragmento diferente para "Pendientes de Asignar" y "Asignadas".
 */
class DashboardPagerAdapter(fa: FragmentActivity) : FragmentStateAdapter(fa) {

    // Se definen explícitamente las dos pestañas necesarias
    private companion object {
        const val NUM_TABS = 2
    }

    override fun getItemCount(): Int = NUM_TABS

    /**
     * Devuelve la instancia del Fragmento correspondiente a la posición de la pestaña.
     */
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> PendingRequestsFragment() // ⬅️ Pestaña 0: Solicitudes Pendientes (para asignación)
            1 -> AssignedRequestsFragment() // ⬅️ Pestaña 1: Solicitudes Asignadas (para seguimiento)
            else -> throw IllegalStateException("Posición de pestaña inválida: $position")
        }
    }
}